app/main.py

```diff
@@ -1,3 +1,61 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (float, int)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (float, int)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: float | int) -> Distance:
+        return Distance(round(self.km / other, 2))
```

Comment:
There is no check for division by zero in the `__truediv__` method. This can lead to a `ZeroDivisionError` at runtime. Consider adding a check to ensure `other` is not zero before performing the division.